### PR TITLE
fix: Avatar doesn't scale when display is none

### LINKED ISSUE #26521: Avatar组件在display:none时不会计算缩放
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/pensive-brown-ns9nl?file=/index.js)

### Steps to reproduce
点击Toggle按钮

### What is expected?
正确计算缩放

### What is actually happening?
没计算缩放

| Environment | Info |
|---|---|
| antd | 4.6.2 |
| React | 16.13.1 |
| System | macOS 10.15.6 |
| Browser | Chrome 84 |

---
看了下在v3中每次update都有计算判断，现在的没有每次update都计算

### PR DESCRIPTION
**Pull request by zhangyu1818**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #26521 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Avatar doesn't scale fallback text well when display is none. |
| 🇨🇳 Chinese | 修复 Avatar 在 `display: none` 时不会正确缩放 fallback 文字的问题。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
来个用例？

File: components/avatar/avatar.tsx [hunk 1]
```diff
@@ -79,13 +79,7 @@ const InternalAvatar: React.ForwardRefRenderFunction<unknown, AvatarProps> = (pr
 
   React.useEffect(() => {
     setScaleParam();
-  }, [props.children, props.gap, props.size]);
-
-  React.useEffect(() => {
-    if (props.children) {
-      setScaleParam();
-    }
-  }, [isImgExist]);
```
Comment (zombieJ) [inline components/avatar/avatar.tsx:88]: 每次 effect 都算一次太耗费了，包一个 ResizeObserver 吧

**Comment by zhangyu1818** [comment]
不知道为啥Popconfirm的测试没过，大佬能帮我看看嘛

File: components/avatar/avatar.tsx (hunk 1 above)
Comment (zhangyu1818) [reply components/avatar/avatar.tsx:88]: 大佬，这里改好了，但是不知道为啥Popconfirm的测试没过

**Comment by zhangyu1818** [comment]
这个测试好像不好写，我看见之前偏右大佬专门写了一个demo文件来测试这个问题

**Pull request merged**